resources/arduino_files/camera_board/camera_board.ino

```diff
@@ -5,162 +5,261 @@
+
+      // Provide a visible signal that setup has been entered.
+      if (Serial) {
+        // 2 seconds of fast blinks.
+        for (int i = 0; i < 40; ++i) {
+          delay(50);
+          toggle_led();
+        }
+        Serial.println("LED blink complete");
+      } else {
+        // 2 seconds of slow blinks.
+        for (int i = 0; i < 10; ++i) {
+          delay(200);
+          toggle_led();
+        }
+      }
+    }
+
+    void update() {
+      unsigned long now = millis();
+      if (next_change_ms_ <= now) {
+        toggle_led();
+        next_change_ms_ += (Serial ? 1000 : 100);
+        if (next_change_ms_ <= now) {
+          next_change_ms_ = now;
+        }
+      }
+    }
+
+  private:
+    unsigned long next_change_ms_ = 0;
+} led_handler;
 
 void setup(void) {
   Serial.begin(9600);
   Serial.flush();
 
-  // Turn off LED inside camera box
-  pinMode(LED_BUILTIN, OUTPUT);
-  digitalWrite(LED_BUILTIN, LOW);
+  led_handler.init();
 
   // Setup Camera relays
-  pinMode(CAM_01_RELAY, OUTPUT);
-  pinMode(CAM_02_RELAY, OUTPUT);
-
-  pinMode(RESET_PIN, OUTPUT);  
+  pinMode(CAM_0_RELAY, OUTPUT);
+  pinMode(CAM_1_RELAY, OUTPUT);
 
   // Turn on Camera relays
-  turn_pin_on(CAM_01_RELAY);
-  turn_pin_on(CAM_02_RELAY);
-
-  if (! accelerometer.begin()) {
-    while (1);
-  }
+  turn_pin_on(CAM_0_RELAY);
+  turn_pin_on(CAM_1_RELAY);
 
-  dht.begin();
+  acc_handler.init();
+  dht_handler.init();
 
-  // Check Accelerometer range
-  // accelerometer.setRange(MMA8451_RANGE_2_G);
-  // Serial.print("Accelerometer Range = "); Serial.print(2 << accelerometer.getRange());
-  // Serial.println("G");
+  Serial.println("EXIT setup()");
+  next_report_millis = end_setup_millis = millis();
 }
 
 void loop() {
-  
+  led_handler.update();
+  if (Serial) {
+    main_loop();
+  }
+}
+
+static int inputs = 0;
+
+void main_loop() {
+  unsigned long now = millis();
+  if (next_report_millis <= now) {
+    // Schedule the next report for `interval' milliseconds from the last report,
+    // unless we've fallen behind.
+    constexpr int interval = 1000;
+    next_report_millis += interval;
+    if (next_report_millis <= now) {
+      next_report_millis = now + interval;
+    }
+
+    // Collect the data. Since some of these operations take a while, keep updating the
+    // LED as appropriate. Could probably be done with an interrupt handler instead.
+    report_num++;
+    acc_handler.collect();
+    led_handler.update();
+    dht_handler.collect();
+    led_handler.update();
+    bool cam0 = digitalRead(CAM_0_RELAY);
+    bool cam1 = digitalRead(CAM_1_RELAY);
+
+    // Format/output the results.
+    Serial.print("{\"name\":\"camera_board\", \"count\":");
+    led_handler.update();
+    Serial.print(millis() - end_setup_millis);
+    led_handler.update();
+    Serial.print(", \"num\":");
+    led_handler.update();
+    Serial.print(report_num);
+    led_handler.update();
+    Serial.print(", \"inputs\":");
```

Comment:
`inputs` was not at all clear to me until reading below. Maybe name the key `inputs_recvd` or something similar.

Reply:
OK